File: motor3d 2.0.ConvertedToC#/PosicionRelativa3D.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
namespace Motor3D.Espacio3D
{
	public enum TipoPosicionRelativa3D
	{
		Secante,
		Coincidente,
		Paralelo,
		Cruce
	}

	public class PosicionRelativa3D : ObjetoGeometrico3D
	{

		private TipoPosicionRelativa3D mTipo;

		private Punto3D mInterseccion;
		public TipoPosicionRelativa3D Tipo {
			get { return mTipo; }
		}

		public Punto3D Interseccion {
			get {
				if (mTipo == TipoPosicionRelativa3D.Secante) {
					return mInterseccion;
				} else {
					throw new ExcepcionGeometrica3D("POSICIONRELATIVA3D (INTERSECCION_GET): La posición no es de tipo secante. No se puede obtener la intersección." + Constants.vbNewLine + "Tipo=" + mTipo.ToString());
				}
			}
		}

		public PosicionRelativa3D(Punto3D ValInterseccion)
		{
			mTipo = TipoPosicionRelativa3D.Secante;
			mInterseccion = ValInterseccion;
		}

		public PosicionRelativa3D(TipoPosicionRelativa3D ValTipoPosicion)
		{
			mTipo = ValTipoPosicion;
		}

		public override string ToString()
		{
			if (mTipo == TipoPosicionRelativa3D.Secante) {
				return "{" + mTipo.ToString() + ",Intersección=" + mInterseccion.ToString() + "}";
			} else {
				return "{" + mTipo.ToString() + "}";
			}
		}
	}
}

File: motor3d 2.0/posicionrelativa3d.vb
Namespace Espacio3D
    Public Enum TipoPosicionRelativa3D
        Secante
        Coincidente
        Paralelo
        Cruce
    End Enum

    Public Class PosicionRelativa3D
        Inherits ObjetoGeometrico3D

        Private mTipo As TipoPosicionRelativa3D
        Private mInterseccion As Punto3D

        Public ReadOnly Property Tipo() As TipoPosicionRelativa3D
            Get
                Return mTipo
            End Get
        End Property

        Public ReadOnly Property Interseccion() As Punto3D
            Get
                If mTipo = TipoPosicionRelativa3D.Secante Then
                    Return mInterseccion
                Else
                    Throw New ExcepcionGeometrica3D("POSICIONRELATIVA3D (INTERSECCION_GET): La posición no es de tipo secante. No se puede obtener la intersección." & vbNewLine _
                                                    & "Tipo=" & mTipo.ToString)
                End If
            End Get
        End Property

        Public Sub New(ByVal ValInterseccion As Punto3D)
            mTipo = TipoPosicionRelativa3D.Secante
            mInterseccion = ValInterseccion
        End Sub

        Public Sub New(ByVal ValTipoPosicion As TipoPosicionRelativa3D)
            mTipo = ValTipoPosicion
        End Sub

        Public Overrides Function ToString() As String
            If mTipo = TipoPosicionRelativa3D.Secante Then
                Return "{" & mTipo.ToString & ",Intersección=" & mInterseccion.ToString & "}"
            Else
                Return "{" & mTipo.ToString & "}"
            End If
        End Function
    End Class
End Namespace
